### ISSUE TITLE
How to use with TensorFlow dataset API?

### ISSUE THREAD

**Issue by Spenhouet** [bethgelab/foolbox#262]
The current `TensorFlowModel` seems to expect a placeholder and to feed this placeholder within the `attack` method.

The problem is, by using the TensorfFlow dataset API there will be no placeholder.
The tensor (iterator) provided will be feed the correct samples by the dataset API.

Is foolbox unusable for TensorFlow models using the dataset API?

If not, could you give an example?

I currently try something like this:

```
train_dataset = ...
test_dataset = ...

dataset_iter = tf.data.Iterator.from_structure(...)
train_initializer = dataset_iter.make_initializer(train_dataset)
test_initializer = dataset_iter.make_initializer(test_dataset)

features, labels = dataset_iter.get_next()

#----------               

import foolbox

model = foolbox.models.TensorFlowModel(features, logits, bounds=(0, 1))
criterion = foolbox.criteria.Misclassification()
attack = foolbox.attacks.LBFGSAttack(model, criterion)

image, label = sess.run([features[0], tf.argmax(labels[0])])

adversarial = attack(image, label)
```

which obviously fails since `features` is a iterator and not a placeholder.

If there is currently no support for the dataset API, I propose the following:

```
model = foolbox.models.TensorFlowModel(features, logits, labels) 
```

The parameters `features` and `labels` are dataset API iterators.
The `attack` method doesn't require any parameter since features and labels are already provided by the `TensorFlowModel`.

_(I omitted some implementation details to better convey the idea)_

More on the dataset API: https://www.tensorflow.org/guide/datasets

**Comment by Spenhouet** [comment]
I resolved my issue.

**Comment by wayneowen7** [comment]
Hi,i got the same problem !!(my external model was inplement by dataset API),but it's hard for me to comecross the problem ,could you share you solution or just for a hint?

**Comment by wayneowen7** [comment]
@Spenhouet
